alibi_detect/cd/utils.py

```diff
@@ -39,3 +43,67 @@ def update_reference(X_ref: np.ndarray,
             raise KeyError('Only `reservoir_sampling` and `last` are valid update options for X_ref.')
     else:
         return X_ref
+
+
+def activate_train_mode_for_dropout_layers(model: Union[nn.Module, tf.keras.Model], backend: str) -> Callable:
+    # TODO: Figure a way to do this properly for tensorflow models.
+    if backend == 'pytorch':
+        model.eval()
+        n_dropout_layers = 0
+        for module in model.modules():
+            if isinstance(module, nn.Dropout):
+                module.train()
+                n_dropout_layers += 1
+        if n_dropout_layers == 0:
+            raise ValueError("No dropout layers identified.")
+    elif backend == 'tensorflow':
+        model.trainable = False
+        model = partial(model, training=True)  # Note this affects batchnorm etc also
+    else:
+        raise NotImplementedError("Only 'pytorch' or 'tensorflow' backends supported")
+
+    return model
+
+
+def get_preds(
+    x: np.ndarray,
+    model: Callable,
+    backend: str,
+    batch_size: int,
+    shuffle: bool = False,
+    force_full_batches: bool = False,
+    device: Optional[str] = None,
+    tokenizer: Optional[Callable] = None,
+    max_len: Optional[int] = None,
+) -> np.ndarray:
+
+    backend = backend.lower()
+    model_kwargs = {
+        'model': model, 'batch_size': batch_size, 'tokenizer': tokenizer, 'max_len': max_len
+    }
+    if backend == 'tensorflow':
```

Comment:
For backend specification, best to have checks in place, see eg https://github.com/SeldonIO/alibi-detect/blob/master/alibi_detect/cd/classifier.py#L98. You could practically copy the check linked over without the ClassifierDrift reerence and in a follow up PR we could implement a `_check_backend` function that you would use everywhere where dispatching occurs. @jklaise ?

Reply:
Yea we should factor out `_validate_backend` or similar, can you open an issue?